vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -28,13 +34,18 @@ def get_min_capability(cls) -> int:
         # be included: https://github.com/pytorch/pytorch/pull/118881
         return 90
 
+    @classmethod
+    def require_config_file(cls) -> bool:
+        return False
+
     @classmethod
     def get_config_filenames(cls) -> List[str]:
-        return []
+        return ["quantize_config.json"]
```

Comment:
Just wondering: For static quantization, can we just have this scaling factor in `config.json`? I'm not sure if this is a better decision than having a separate quantization config file, but it seems feasible. WDYT?

Reply:
Yes it is feasible. The reason why I put it into the safetensor files is because that's a better place to store tensors than the config.json and for some schemes, these tensors can be larger (e.g. for per-channel quantization). Since the checkpoint already needs to be rewritten (to add the quant_config.json and also possibly to convert weights to FP8), I don't think it is a big problem to rewrite the safetensors to include activation scales. This is also what https://huggingface.co/FriendliAI/Mistral-7B-Instruct-v0.2-fp8 does (note that their model format is otherwise not very useful since it stores the weights as int8).

I assume that a standard for FP8 will emerge, and I would expect it to store the scales in the safetensor files -- this is a no brainer for weight scales but also makes sense for activation scales. Once we have a standard, we should use that. Right now, I don't think trying to invent our own standard in `quantize_config.json` is a good idea (since it involves a schema), whereas storing scales in the safetensor scales is pretty canonical.

These are my reasons -- let me know if you prefer otherwise :)

Reply:
Also cc @robertgshaw2-neuralmagic who also thought about this a bunch I think :)

Reply:
I agree with @pcmoritz , the `scales` should be in a safetensors file

The mental model for the `quantize_config.json` is that it would hold metadata about what is in the `safetensors` file.

So examples could be:
- Datatype of the weights
- Whether the activations are static or dynamic (so we dont have to peek into safetensors)
- Channelwise vs not, etc

For this first implementation, we don't this, but if we start supporting various different schemes, then we will (since we need to know this when `create_weights` is called - which happens before we look at the `safetensors` file)

Reply:
Also, mostly `quantize_config` is not used anymore, even for `AutoGPTQ` 

The `config.json` typically has a `quantization_config`. 

https://huggingface.co/astronomer/Llama-3-8B-Instruct-GPTQ-8-Bit/blob/main/config.json

We only fall back to `quantize_config.json` if `quantization_config` is not found in the `config.json`